rclcpp/test/rclcpp/test_parameter_client.cpp

```diff
@@ -291,3 +291,43 @@ TEST_F(TestParameterClient, sync_parameter_get_parameter_types) {
   ASSERT_EQ(1u, parameter_types.size());
   ASSERT_EQ(rclcpp::ParameterType::PARAMETER_INTEGER, parameter_types[0]);
 }
+
+/*
+  Coverage for async describe_parameters
+ */
+TEST_F(TestParameterClient, async_parameter_describe_parameters) {
+  auto asynchronous_client = std::make_shared<rclcpp::AsyncParametersClient>(node);
+  bool callback_called = false;
+  auto callback = [&callback_called](
+    std::shared_future<std::vector<rcl_interfaces::msg::ParameterDescriptor>> result)
+    {
+      // We expect the result to be empty since we tried to get a parameter that didn't exist.
+      if (result.valid() && result.get().size() == 0) {
+        callback_called = true;
+      }
+    };
+  std::vector<std::string> names{"foo"};
+  std::shared_future<std::vector<rcl_interfaces::msg::ParameterDescriptor>> future =
+    asynchronous_client->describe_parameters(names, callback);
+  auto return_code = rclcpp::spin_until_future_complete(
+    node, future, std::chrono::milliseconds(100));
+  ASSERT_EQ(rclcpp::FutureReturnCode::SUCCESS, return_code);
+  ASSERT_TRUE(callback_called);
+}
+
+/*
+  Coverage for sync describe_parameters
+ */
+TEST_F(TestParameterClient, sync_parameter_describe_parameters) {
+  node->declare_parameter("foo", 4);
+  auto synchronous_client = std::make_shared<rclcpp::SyncParametersClient>(node);
+  std::vector<std::string> names{"foo"};
+  std::vector<rcl_interfaces::msg::ParameterDescriptor> parameter_types =
+    synchronous_client->describe_parameters(names);
+  ASSERT_EQ(1u, parameter_types.size());
```

Comment:
And similarly, can we add a test for the `SyncParameterClient` where the parameters don't exist, and also if some exist and some don't?